Task 4/Task4/Task4/CastomGit.cs

```diff
@@ -0,0 +1,150 @@
+﻿using Actions;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+
+namespace Task4
+{
+    class CastomGit
+    {
+        private string path;
+
+        private List<(DateTime, FileAction)> Log = new List <(DateTime, FileAction)>();
+
+        public Dictionary<String, String> FilePastPath = new Dictionary<string, string>();
+
+        private UserCommunication communicator = new UserCommunication();
```

Comment:
Good that you moved user interaction logic to it's own class, but the code is still tightly coupled with your implementation - it's composition, not aggregation (composition is just moving code to a separate class to reduce clutter, while aggregation's point in reusing some other class logic with possibility of using different implementations). Better would be to create an interface and pass such communicator in ctor.